Add option to clear the terminal by pressing "c"

Levi-Lesches opened the pull request:
## Description

When using commands like `flutter run` and `flutter attach` on the terminal, pressing "c" will now clear the output. This is very helpful for when errors and other outputs clog the screen with irrelevant information. 

## Related Issues

 #38819  

## Tests

I don't know how to add tests for terminal behavior, but the functionality is pretty simple, just print the `clear` character (already defined in some other file). 

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: *Replace with a link to your migration guide*

<!-- Links -->
[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

Levi-Lesches committed:
Add option to clear the terminal

Levi-Lesches committed:
Merge branch 'master' of https://github.com/flutter/flutter

Levi-Lesches committed:
Merge branch 'master' of https://github.com/flutter/flutter

Levi-Lesches commented:
Yeah can anyone tell me if I need tests? Or how to write them? It's just a print statement and an extra `case`.

jonahwilliams reviewed (commented):
Seems like a reasonable approach, but yes you'll absolutely need tests. See some of the existing test cases in the resident_runner test files

Levi-Lesches committed:
Merge branch 'master' of https://github.com/flutter/flutter

Levi-Lesches committed:
Merge https://github.com/Levi-Lesches/flutter

zanderso commented on packages/flutter_tools/lib/src/resident_runner.dart:
```diff
@@ -1052,16 +1052,16 @@ abstract class ResidentRunner {
       commandHelp.P.print();
       commandHelp.a.print();
     }
-    if (flutterDevices.any((FlutterDevice d) => d.device.supportsScreenshot)) {
```
I think this shouldn't be deleted.

Levi-Lesches replied:
It appears twice, just a few lines before this one (sorry, I meant to mention that), but I can put it in another PR if it'll make conflicts.

zanderso replied:
Oh, you're right. I wonder how that happened.

gspencergoog committed:
Revert "Always use new plugin schema for plugin template (#48258)" (#49827)

This reverts commit 9d4a35b because it appears to break

the gradle_non_android_plugin_test.

Levi-Lesches commented:
Do you happen to know what those tests failures are and how to fix them? They didn't happen when I ran the tests on my computer, and the lines I added should not cause build commands to fail...

zanderso commented:
This list:
https://github.com/flutter/flutter/blob/6ee8b6546ba4a5b2dbed57a88d88aa074223fccf/packages/flutter_tools/test/general.shard/resident_runner_test.dart#L375
Needs to have the new option added to it.

gspencergoog committed:
Use DiagnosticableMixin instead of Diagnosticable for some conditionals. (#49726)

This PR fixes #49647 by changing DiagnosticableMixin.toDiagnosticsNode and other things to accept DiagnosticableMixin instead of requiring Diagnosticable.

yjbanov committed:
Initial web benchmark harness and 3 benchmarks (#49460)

Initial web benchmark harness and 3 benchmarks

chunhtai committed:
Refactors global key duplication detection (#46183)

Commit:
Revert "Refactors global key duplication detection (#46183)" (#49847)

This reverts commit 340b193f20e85f2cfbebf84fe315112985a38a13.

engine-flutter-autoroll committed:
Roll engine 74d07b5f6b4f..e5209202fb00 (4 commits) (#49835)


https://github.com/flutter/engine/compare/74d07b5f6b4f...e5209202fb00

git log 74d07b5f6b4f..e5209202fb00 --first-parent --oneline
2020-01-30 skia-flutter-autoroll@skia.org Roll src/third_party/dart fc3af737c759..162d6c5634a0 (209 commits) (flutter/engine#16261)
2020-01-30 skia-flutter-autoroll@skia.org Roll src/third_party/skia d1be5d64f8a7..36c0521d57de (13 commits) (flutter/engine#16260)
2020-01-30 mouad.debbar@gmail.com [web] Correct getPositionForOffset for multi-line paragraphs (flutter/engine#16206)
2020-01-30 skia-flutter-autoroll@skia.org Roll fuchsia/sdk/core/linux-amd64 from -mGIA... to 93K0d... (flutter/engine#16257)


If this roll has caused a breakage, revert this CL and stop the roller
using the controls here:
https://autoroll.skia.org/r/flutter-engine-flutter-autoroll
Please CC cbracken@google.com on the revert to ensure that a human
is aware of the problem.

To report a problem with the AutoRoller itself, please file a bug:
https://bugs.chromium.org/p/skia/issues/entry?template=Autoroller+Bug

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+/master/autoroll/README.md

GaryQian committed:
Track lastKnownRemoteTextEditingValue separately from received data (#49406)

a14n committed:
Revert "Roll engine 74d07b5f6b4f..e5209202fb00 (4 commits) (#49835)" (#49869)

This reverts commit 25f798a1aaaa292c1e30d25b7a6cc54fed38ab45.

a14n committed:
refactor generateString to emit simpler dart code (#49334)

engine-flutter-autoroll committed:
Roll engine 74d07b5f6b4f..804dca62a17c (13 commits) (#49873)


https://github.com/flutter/engine/compare/74d07b5f6b4f...804dca62a17c

git log 74d07b5f6b4f..804dca62a17c --first-parent --oneline
2020-01-31 yjbanov@google.com Use bundled Roboto in all tests (flutter/engine#16218)
2020-01-31 gw280@google.com Revert "Migrate flutter_runner from flutter_runner::{Thread,Loop} to fml::{Thread,MessageLoop} (#15118)" (flutter/engine#16277)
2020-01-31 skia-flutter-autoroll@skia.org Roll src/third_party/skia 36c0521d57de..6305b2f8342a (8 commits) (flutter/engine#16272)
2020-01-31 jason-simmons@users.noreply.github.com Revert "[web] Correct getPositionForOffset for multi-line paragraphs (#16206)" (flutter/engine#16268)
2020-01-30 stuartmorgan@google.com Fix Windows file checks of unicode paths (flutter/engine#16105)
2020-01-30 iska.kaushik@gmail.com [fuchsia] Fix the import for dart_api.h (flutter/engine#16269)
2020-01-30 iska.kaushik@gmail.com [fuchsia] SceneHostBindings are no longer thread locals (flutter/engine#16262)
2020-01-30 gw280@google.com Pass through invoker.resources in fuchsia_test_archive (flutter/engine#16265)
2020-01-30 matthew-carroll@users.noreply.github.com Notify PlatformViewsController within FlutterEngine when a hot restart occurs. (#48518) (flutter/engine#16230)
2020-01-30 skia-flutter-autoroll@skia.org Roll src/third_party/dart fc3af737c759..162d6c5634a0 (209 commits) (flutter/engine#16261)
2020-01-30 skia-flutter-autoroll@skia.org Roll src/third_party/skia d1be5d64f8a7..36c0521d57de (13 commits) (flutter/engine#16260)
2020-01-30 mouad.debbar@gmail.com [web] Correct getPositionForOffset for multi-line paragraphs (flutter/engine#16206)
2020-01-30 skia-flutter-autoroll@skia.org Roll fuchsia/sdk/core/linux-amd64 from -mGIA... to 93K0d... (flutter/engine#16257)


If this roll has caused a breakage, revert this CL and stop the roller
using the controls here:
https://autoroll.skia.org/r/flutter-engine-flutter-autoroll
Please CC cbracken@google.com on the revert to ensure that a human
is aware of the problem.

To report a problem with the AutoRoller itself, please file a bug:
https://bugs.chromium.org/p/skia/issues/entry?template=Autoroller+Bug

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+/master/autoroll/README.md

Levi-Lesches commented:
Oh thank you -- I was looking for that since I saw it in the tests.

dnfield committed:
Elide tree walks (#48413)

Levi-Lesches committed:
Merge branch 'master' into clear

Commit:
Revert "Elide tree walks (#48413)" (#49886)

This reverts commit 46ccd086745bb9ade53f2457f3bb91942b2af970.

Levi-Lesches commented:
I'm not quite sure what happened but apparently the tests are broken before I even touched them (apparently `globals` does not actually define `fs`), so I'll try to fix them and hope my changes don't conflict with anyone else's.

Levi-Lesches commented:
After doing a git pull a LOT is broken in the files I'm trying to modify... so I'll wait for those to be fixed before continuing, since I'm assuming someone else is working on it in another PR.

dnfield committed:
Be clearer about when and why we override HttpClient in tests (#49844)

2ZeroSix committed:
fix material selection handle paint with transparent textSelectionHandleColor  (#49830)

a14n committed:
remove unnecessary string interpolations (#49622)

mannprerak2 committed:
switching channel prompts to run flutter upgrade (#49511)

jonahwilliams commented:
@Levi-Lesches all of the tool tests are currently green at head. What sorts of errors are you running into with the tests?

Levi-Lesches committed:
Added test for clearing the screen

Levi-Lesches committed:
Merge branch 'master' into clear

Commit:
Use skia golden files in driver test (#49750)

Levi-Lesches committed:
Merge branch 'clear'

Levi-Lesches commented:
I added one test to see if running `ResidentRunner.clearScreen()` sets `testLogger.statusText` to `""` and it passed, and I added the command to the help section, so is everything good? If not, I can add more, but I'm not sure what else is needed.

jonahwilliams commented:
@Levi-Lesches definitely don't rebase like that :( . Can you revert that commit please?

Levi-Lesches commented:
Sorry, I haven't really used Git collaboratively yet. Which commit are you referring to?

jonahwilliams commented:
A general workflow I recommend:

1. Update upstream: `git fetch upstream`

2. Branch from master: `git checkout upstream/master -b my_feature_branch`

3. Make changes to local branch: `git commit`

4. Prepare for PR: `git push origin my_feature_branch`

To pull in the latest master:

1. `git fetch upstream`

2. `git pull upstream master`

3. `git push origin my_feature_branch`

Levi-Lesches commented:
I've been doing that, but accidentally created the feature branch *after* I made this PR, so I merged it to my own master (because that's where this PR is referencing). Should I start a new PR from a feature branch instead?

Also, which particular commit where you referring to that rebased weirdly? I didn't explicitly rebase, just merged my master and feature branches.

jonahwilliams commented:
All of the commits that aren't yours: https://github.com/flutter/flutter/pull/49819/commits

Levi-Lesches commented:
Hm, yeah those definitely shouldn't be there. I'll try to clean things up -- shoud I submit a new PR so I can have it come from my own feature branch?

jonahwilliams commented:
In my experience its often easier to start a new PR rather than make git behave...

Levi-Lesches commented:
Made a new PR in #50034.

jonahwilliams commented:
Okay, I'll take a look. Closing this one!
